lib/bamboo/adapters/send_grid_adapter.ex

```diff
@@ -148,6 +152,14 @@ defmodule Bamboo.SendGridAdapter do
 
   defp put_reply_to(body, _), do: body
 
+  defp put_headers(body, %Email{headers: headers}) when is_map(headers) do
+    prepared_headers = Map.delete(headers, "reply-to")
```

Comment:
From the [Sendgrid docs](https://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/index.html):
```
Must not be any of the following reserved headers: x-sg-id, x-sg-eid, received, dkim-signature, Content-Type, Content-Transfer-Encoding, To, From, Subject, Reply-To, CC, BCC
```

Would it be worth it to explicitly delete these reserved keys (e.g. `Subject`) here? Or log a warning if they're passed? Or just link to the Sendgrid documentation on reserved headers in the `moduledoc`?

Reply:
@michaelm244 Thanks for the PR! This is a great question. Do we know what happens if you try to send an email with one of these reserved headers? Does it just get ignored? I like the idea of logging something to notify the user, and definitely linking to the docs. If it just gets ignored it may not be necessary to delete it, but if there's some other effect or if it causes an error then we would want to delete it. What are your thoughts?

Reply:
Just tested out by adding a header with the key of `Subject` - Sendgrid returns a 500 response with a very informative error message:
```
{
  "errors": [
    {
      "message": "Header can not be one of the reserved keys. Refer to documentation link.",
      "field": "headers.Subject",
      "help": "http://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/errors.html#message.headers"
    }
  ]
}
```

Since the Sendgrid API enforces it well, is it necessary for this library to enforce it also? The user will have very clear feedback of what's going wrong if they try to send a disallowed header. If the library enforces it, it will also take on the burden of ensuring that the library's list of disallowed headers is always up-to-date with Sendgrid's.

Reply:
@maymillerricci also just linked to Sendgrid docs in the `@moduledoc`

Reply:
@michaelm244 So sorry for the delay! Yes, I agree that since the user will get clear feedback then we don't need to enforce it on this end, and we don't want the extra burden of keeping the lists lined up. This looks great! My only other thought is: do you want to add a test for this special case of the `reply-to` header that's being deleted since it is handled elsewhere?